src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,145 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.AddCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+import duke.command.DoneCommand;
+import duke.exception.DukeException;
+import duke.exception.ExtraDescriptionException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidNumberException;
+
+/**
+ * Parser class.
+ */
+class Parser {
+    private static String[] oneLine;
+
+    /**
+     * check str is a number.
+     *
+     * @param str The str to be checked.
+     * @return Ture if str is a number, otherwise False.
+     */
+    private boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * feature of find command.
+     *
+     * @return FindCommand object.
+     */
+    private FindCommand findFeature() throws DukeException {
+        if (oneLine.length == 1 || (oneLine.length == 2 && oneLine[1].isBlank())) {
```

Comment:
Perhaps it might be easier for understanding to abstract the condition into another method that returns a boolean so it is more readable? Same for the other methods below